vim-patch:8.1.2227,8.2.{315,928,1007,1052,1121,1580,2221,2229,2231,2232}

janlazo opened the pull request.

janlazo committed:
vim-patch:8.2.2221: if <Down> is mapped on the command line 'wildchar' is inserted

Problem:    If <Down> is mapped on the command line 'wildchar' is inserted.
Solution:   Set KeyTyped when using 'wildchar'. (closes vim/vim#7552)
https://github.com/vim/vim/commit/b0ac4ea5e1c5f0ff4e951978c32ccfffe46916f8

bfredl commented on src/nvim/ex_getln.c:
```diff
@@ -1082,6 +1082,7 @@ static int command_line_execute(VimState *state, int key)
     if (s->c == K_DOWN && ccline.cmdpos > 0
         && ccline.cmdbuff[ccline.cmdpos - 1] == '.') {
       s->c = (int)p_wc;
```
Hmm, `Ctrl_Z` should always work in neovim without needing to fudge `KeyTyped`.

janlazo replied:
Ok. I'll test in a separate commit.

janlazo replied:
It doesn't work.

janlazo commented on src/nvim/testdir/test_cmdline.vim:
```diff
@@ -51,6 +51,19 @@ func Test_complete_wildmenu()
 
+  +  " <C-J>/<C-K> mappings to go up/down directories when 'wildcharm' is
+  " different than 'wildchar'.
+  set wildcharm=<C-Z>
```
`wildcharm` option is set to non-zero so https://github.com/neovim/neovim/commit/bed9839f46047abd81f7c9b56936ba177829221c won't work here.

bfredl replied:
bed9839 should always work as it is enabled unconditionally.

janlazo replied:
`wildchar` is also non-zero by default.

bfredl replied:
still, bed9839 keeps working.

janlazo replied:
The test fails without the C code changes.

janlazo committed:
vim-patch:8.1.2227: layout wrong if 'lines' changes while cmdline window is open

Problem:    Layout wrong if 'lines' changes while cmdline window is open.
Solution:   Do not restore the window layout if 'lines' changed.
            (closes vim/vim#5130)
https://github.com/vim/vim/commit/1c329c04be2e95a3589a53f2978926e91b450cca

janlazo committed:
vim-patch:8.2.1007: completion doesn't work after ":r ++arg !"

Problem:    Completion doesn't work after ":r ++arg !".
Solution:   Skip over "++arg". (Christian Brabandt, closes vim/vim#6275,
            closes vim/vim#6258)
https://github.com/vim/vim/commit/c8cb883015619cfe6db931ac2d6e15b0be5c56ed

janlazo committed:
vim-patch:8.2.1121: command completion not working after ++arg

Problem:    Command completion not working after ++arg.
Solution:   Move skipping up. (Christian Brabandt, closes vim/vim#6382)
https://github.com/vim/vim/commit/743d0620203388bf87dc611cea544b485e4b9f85

janlazo committed:
vim-patch:8.2.1580: wildmenu does not work properly

Problem:    Wildmenu does not work properly.
Solution:   Do not call may_do_incsearch_highlighting() if completion is in
            progress.
https://github.com/vim/vim/commit/a60053b8f4cc7e135ba9496a8f4855d26aee09e7

janlazo committed:
vim-patch:8.2.0928: many type casts are used for vim_strnsave()

Problem:    Many type casts are used for vim_strnsave().
Solution:   Make the length argument size_t instead of int. (Ken Takata,
            closes vim/vim#5633)  Remove some type casts.
https://github.com/vim/vim/commit/df44a27b53586fccfc6a3aedc89061fdd9a515ff

N/A patches for version.c:

vim-patch:8.2.0315: build failure on HP-UX system

Problem:    Build failure on HP-UX system.
Solution:   Use LONG_LONG_MIN instead of LLONG_MIN.  Add type casts for switch
            statement. (John Marriott)
https://github.com/vim/vim/commit/c593bec4120f122e8a9129ec461968f1bd214435

vim-patch:8.2.1052: build failure with older compilers

Problem:    Build failure with older compilers.
Solution:   Move declaration to start of block.
https://github.com/vim/vim/commit/7acde51832f383f9a6d2e740cd0420b433ea841a

vim-patch:8.2.2229: build failure without the +eval feature

Problem:    build failure without the +eval feature.
Solution:   Add #ifdef.
https://github.com/vim/vim/commit/39cb2dab18e85fc60f116a4543e433616872b690

vim-patch:8.2.2232: compiler error for falling through into next case

Problem:    Compiler error for falling through into next case.
Solution:   Move FALLTHROUGH below the #endif
https://github.com/vim/vim/commit/9618a25b9c054f0ee4e267d2db96b6e7c113ed7a

janlazo committed:
vim-patch:8.2.2231: when "--remote file" is used "file" is not reloaded

Problem:    When "--remote file" is used "file" is not reloaded.
Solution:   When a :drop command is used for a file that is already displayed
            in a window and it has not been changed, check if it needs to be
            reloaded. (closes vim/vim#7560)
https://github.com/vim/vim/commit/e4862a0fe62261754daf476866ef2aa8586b716c

Remove unused "focus" parameter from "buf_check_timestamp()".
It was meant for removed GUI code.
